prow/cmd/tot/main.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Tot vends incrementing numbers for use in builds.
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+var (
+	port        = flag.Int("port", 8888, "Port to listen on.")
+	logJson     = flag.Bool("log-json", false, "output log in JSON format")
+	storagePath = flag.String("storage", "tot.json", "Where to store the results")
+)
+
+type store struct {
+	Number      map[string]int
+	mutex       sync.Mutex
+	storagePath string
+}
+
+func newStore(storagePath string) (*store, error) {
+	s := &store{
+		Number:      make(map[string]int),
+		storagePath: storagePath,
+	}
+	buf, err := ioutil.ReadFile(storagePath)
+	if err == nil {
```

Comment:
I think you can either explicitly check using `os.IsNotExist(err)`, otherwise you might be ignoring a real error.